Doc/library/socket.rst

```diff
@@ -595,6 +595,45 @@ The following functions all create :ref:`socket objects <socket-objects>`.
    .. versionchanged:: 3.2
       *source_address* was added.
 
+.. function:: bind_socket(address, *, family=AF_UNSPEC, type=SOCK_STREAM, backlog=128, reuse_addr=None, reuse_port=False, flags=None, hybrid_ipv46=False)
+
+   Convenience function which creates a socket bound to *address* (a 2-tuple
+   ``(host, port)``) and return the socket object upon which you can call
+   :meth:`socket.accept()` in order to accept new connections.
+   Internally it relies on :meth:`getaddrinfo()` and returns the first socket
+   which can be bound to *address*.
+   If *host* is an empty string or ``None`` all network interfaces are assumed.
+   If *family* is :data:`AF_UNSPEC` the address family will be determined from
+   the *host* specified in *address*. If family can't clearly be determined
+   from *host* and *hybrid_ipv46* is ``False`` then :data:`AF_INET` family will
+   be preferred over :data:`AF_INET6`.
+   *type* should be either :data:`SOCK_STREAM` or :data:`SOCK_DGRAM`.
+   *backlog* is the queue size passed to :meth:`socket.listen` if
+   :data:`SOCK_STREAM` *type* is used.
+   *reuse_addr* and *reuse_port* dictates whether to use :data:`SO_REUSEADDR`
+   and :data:`SO_REUSEPORT` socket options respectively.
+   *flags* is a bitmask for :meth:`getaddrinfo()`; if ``None``
+   :data:`AI_PASSIVE` is used.
```

Comment:
Few suggested edits:

-----

A convenience function to create a server socket bound to *address*.  The function automates the following:

* resolving *address* with getaddrinfo
* finding the first resolved address that can be listened on
* automatically applying SO_REUSEADDR, if available
* disabling hybrid IPv4 / IPv6 listening
* calling "socket.listen()"

[add a "with-block" example here]

[list all parameters here]

Reply:
Mmm... IMO I think this would be a repetition of what I already describe when talking about the single parameters. I can probably add:

```
   Convenience function which creates a socket bound to *address* (a 2-tuple
   ``(host, port)``) and return the socket object upon which you can call
   :meth:`socket.accept()` in order to accept new connections.
+   It aims at automating all the necessary steps typically needed when creating
+   a server socket.

     [list all parameters here]

+     [add a "with-block" example here]
```

(also note I already added an example later in the doc)

Reply:
The current text has grammatical errors, sentences are too long, and it is overall is a bit hard to follow, IMO.  But up to you, I guess.

Reply:
I changed the doc a bit by:
- adding a paragraph at the top: <<Convenience function which aims at automating all the typical steps needed when creating a server socket>>
- dividing each arg section with a new line
Does that make it a bit more clear? As for the grammatical errors, could you please point them out?